src/models/datasets.js

```diff
@@ -37,6 +37,13 @@ export async function list(config, d2, filters, pagination) {
     ]);
     const listOptions = { fields, filter, page, pageSize, order };
     const collection = await d2.models.dataSets.list(cleanOptions(listOptions));
+    collection.toArray().forEach(dataSet => {
+        const dataInputPeriods = dataSet.dataInputPeriods;
+        if (!_(dataInputPeriods).isEmpty()) {
+            dataSet.startDate = formatDateLong(dataInputPeriods[0].openingDate);
+            dataSet.endDate = formatDateLong(dataInputPeriods[0].closingDate);
+        }
+    });
```

Comment:
Sorry, I should've pointed in the ticket to the (optional) attribute `getValue` of `detailFields`. This way we avoid adding properties in-place to model objects, which can be confusing when using them, and instead pass the details component a getter.

https://github.com/EyeSeeTea/d2-ui-components/blob/master/src/details-box/DetailsBox.component.js#L15

Reply:
No worries, I thought about asking but didn't want to bother you during the weekend :/ Should be ready now